diff --git a/node_modules/@metamorph/react-native-yamap/android/build.gradle b/node_modules/@metamorph/react-native-yamap/android/build.gradle
index 505ea65..a1ea673 100644
--- a/node_modules/@metamorph/react-native-yamap/android/build.gradle
+++ b/node_modules/@metamorph/react-native-yamap/android/build.gradle
@@ -36,6 +36,6 @@ repositories {
 dependencies {
     implementation 'com.google.android.gms:play-services-location:+'
     implementation 'com.facebook.react:react-native:+'
-    implementation 'com.yandex.android:maps.mobile:4.10.0-full'
+    implementation 'com.yandex.android:maps.mobile:4.36.0-full'
     implementation 'androidx.core:core-ktx:1.13.1'
 }
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
index 54d8a0f..828a0e6 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
@@ -21,7 +21,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
 
     override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
         return MapBuilder.builder<String, Any>()
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getExportedCustomBubblingEventTypeConstants(): MutableMap<String, Any>? {
@@ -84,7 +84,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
                     MapBuilder.of("bubbled", "onWorldToScreenPointsReceived")
                 )
             )
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getCommandsMap(): Map<String, Int>? {
@@ -114,7 +114,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
         when (commandType) {
             "setCenter" -> setCenter(
                 castToYaMapView(view),
-                args!!.getMap(0),
+                args!!.getMap(0) ?: return,
                 args.getDouble(1).toFloat(),
                 args.getDouble(2).toFloat(),
                 args.getDouble(3).toFloat(),
@@ -124,11 +124,11 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
 
             "fitAllMarkers" -> fitAllMarkers(view)
             "fitMarkers" -> if (args != null) {
-                fitMarkers(view, args.getArray(0))
+                args.getArray(0)?.let { fitMarkers(view, it) }
             }
 
             "findRoutes" -> if (args != null) {
-                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2))
+                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2) ?: "")
             }
 
             "setZoom" -> if (args != null) {
@@ -152,11 +152,11 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             }
 
             "getScreenPoints" -> if (args != null) {
-                view.emitWorldToScreenPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitWorldToScreenPoints(it, args.getString(1)) }
             }
 
             "getWorldPoints" -> if (args != null) {
-                view.emitScreenToWorldPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitScreenToWorldPoints(it, args.getString(1)) }
             }
 
             else -> throw IllegalArgumentException(
@@ -171,7 +171,8 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
 
     @ReactProp(name = "clusteredMarkers")
     fun setClusteredMarkers(view: View, points: ReadableArray) {
-        castToYaMapView(view).setClusteredMarkers(points.toArrayList())
+        @Suppress("UNCHECKED_CAST")
+        castToYaMapView(view).setClusteredMarkers(points.toArrayList() as ArrayList<Any>)
     }
 
     @ReactProp(name = "clusterColor")
@@ -243,7 +244,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             val vehicles = ArrayList<String>()
             if (jsVehicles != null) {
                 for (i in 0 until jsVehicles.size()) {
-                    vehicles.add(jsVehicles.getString(i))
+                    jsVehicles.getString(i)?.let { vehicles.add(it) }
                 }
             }
             castToYaMapView(view).findRoutes(points, vehicles, id)
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt
index 85be269..f45b0ff 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt
@@ -17,12 +17,12 @@ class YamapCircleManager internal constructor() : ViewGroupManager<YamapCircle>(
     override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getExportedCustomBubblingEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
-            .build()
+            .build().toMutableMap()
     }
 
     @Nonnull
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
index e32d22c..1214544 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
@@ -20,12 +20,12 @@ class YamapMarkerManager internal constructor() : ViewGroupManager<YamapMarker>(
     override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getExportedCustomBubblingEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
-            .build()
+            .build().toMutableMap()
     }
 
     private fun castToMarkerView(view: View): YamapMarker {
@@ -109,7 +109,7 @@ class YamapMarkerManager internal constructor() : ViewGroupManager<YamapMarker>(
             "animatedMoveTo" -> {
                 val markerPoint = args!!.getMap(0)
                 val moveDuration = args.getInt(1)
-                val lon = markerPoint.getDouble("lon").toFloat()
+                val lon = markerPoint?.getDouble("lon")?.toFloat() ?: return
                 val lat = markerPoint.getDouble("lat").toFloat()
                 val point = Point(lat.toDouble(), lon.toDouble())
                 castToMarkerView(view).animatedMoveTo(point, moveDuration.toFloat())
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt
index cf14b19..a958cf9 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt
@@ -18,12 +18,12 @@ class YamapPolygonManager internal constructor() : ViewGroupManager<YamapPolygon
     override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getExportedCustomBubblingEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
-            .build()
+            .build().toMutableMap()
     }
 
     private fun castToPolygonView(view: View): YamapPolygon {
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt
index 8be99c8..c36d3dd 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt
@@ -18,12 +18,12 @@ class YamapPolylineManager internal constructor() : ViewGroupManager<YamapPolyli
     override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getExportedCustomBubblingEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
-            .build()
+            .build().toMutableMap()
     }
 
     private fun castToPolylineView(view: View): YamapPolyline {
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
index 8b05905..c9221f1 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
@@ -21,7 +21,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
 
     override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
         return MapBuilder.builder<String, Any>()
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getExportedCustomBubblingEventTypeConstants(): MutableMap<String, Any>? {
@@ -84,7 +84,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
                     MapBuilder.of("bubbled", "onWorldToScreenPointsReceived")
                 )
             )
-            .build()
+            .build().toMutableMap()
     }
 
     override fun getCommandsMap(): Map<String, Int>? {
@@ -114,7 +114,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
         when (commandType) {
             "setCenter" -> setCenter(
                 castToYaMapView(view),
-                args!!.getMap(0),
+                args!!.getMap(0) ?: return,
                 args.getDouble(1).toFloat(),
                 args.getDouble(2).toFloat(),
                 args.getDouble(3).toFloat(),
@@ -124,11 +124,11 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
 
             "fitAllMarkers" -> fitAllMarkers(view)
             "fitMarkers" -> if (args != null) {
-                fitMarkers(view, args.getArray(0))
+                args.getArray(0)?.let { fitMarkers(view, it) }
             }
 
             "findRoutes" -> if (args != null) {
-                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2))
+                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2) ?: "")
             }
 
             "setZoom" -> if (args != null) {
@@ -152,11 +152,11 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
             }
 
             "getScreenPoints" -> if (args != null) {
-                view.emitWorldToScreenPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitWorldToScreenPoints(it, args.getString(1)) }
             }
 
             "getWorldPoints" -> if (args != null) {
-                view.emitScreenToWorldPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitScreenToWorldPoints(it, args.getString(1)) }
             }
 
             else -> throw IllegalArgumentException(
@@ -237,7 +237,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
 
             if (jsVehicles != null) {
                 for (i in 0 until jsVehicles.size()) {
-                    vehicles.add(jsVehicles.getString(i))
+                    jsVehicles.getString(i)?.let { vehicles.add(it) }
                 }
             }
 
diff --git a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
index e7e9849..cee0b69 100644
--- a/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
+++ b/node_modules/@metamorph/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
@@ -225,7 +225,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         val screenPoints = Arguments.createArray()
 
         for (i in 0 until worldPoints.size()) {
-            val p = worldPoints.getMap(i)
+            val p = worldPoints.getMap(i) ?: continue
             val worldPoint = Point(p.getDouble("lat"), p.getDouble("lon"))
             val screenPoint = mapWindow.worldToScreen(worldPoint)
             screenPoints.pushMap(screenPointToJSON(screenPoint))
@@ -244,7 +244,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         val worldPoints = Arguments.createArray()
 
         for (i in 0 until screenPoints.size()) {
-            val p = screenPoints.getMap(i)
+            val p = screenPoints.getMap(i) ?: continue
             val screenPoint = ScreenPoint(p.getDouble("x").toFloat(), p.getDouble("y").toFloat())
             val worldPoint = mapWindow.screenToWorld(screenPoint)
             worldPoints.pushMap(worldPointToJSON(worldPoint))
@@ -590,10 +590,10 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         if (show) {
             userLocationLayer!!.setObjectListener(this)
             userLocationLayer!!.isVisible = true
-            userLocationLayer!!.isHeadingEnabled = true
+            userLocationLayer!!.isHeadingModeActive = true
         } else {
             userLocationLayer!!.isVisible = false
-            userLocationLayer!!.isHeadingEnabled = false
+            userLocationLayer!!.isHeadingModeActive = false
             userLocationLayer!!.setObjectListener(null)
         }
     }
@@ -685,7 +685,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         val wTransports = Arguments.createMap()
 
         for ((key, value) in transports) {
-            wTransports.putArray(key, Arguments.fromList(value))
+            wTransports.putArray(key, Arguments.fromList(value?.filterNotNull() ?: emptyList<String>()))
         }
 
         routeMetadata.putMap("transports", wTransports)
